source/adapters/hip/command_buffer.cpp

```diff
@@ -284,22 +256,22 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferCreateExp(
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urCommandBufferRetainExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
-  hCommandBuffer->incrementInternalReferenceCount();
-  hCommandBuffer->incrementExternalReferenceCount();
+  hCommandBuffer->incrementReferenceCount();
   return UR_RESULT_SUCCESS;
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL
 urCommandBufferReleaseExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
-  if (hCommandBuffer->decrementExternalReferenceCount() == 0) {
-    // External ref count has reached zero, internal release of created
+  if (hCommandBuffer->decrementReferenceCount() == 0) {
+    // Ref count has reached zero, release of created commands
     // commands.
     for (auto Command : hCommandBuffer->CommandHandles) {
-      commandHandleReleaseInternal(Command);
+      delete Command;
```

Comment:
Since the user is no longer responsible for the lifetime of command handles, would it make sense to use a vector of std::unique_pointer and std::make_unique to manage the lifetime of the object instead of explicit calls to `new`?